File: main_project/message_config/src/java/src/main/java/com/renren/messageconfig/MessageConfigException.java
package com.renren.messageconfig;

/**
 * 异常
 * 
 * @author michael
 * @email liyong@renren-inc.com
 * 
 */
public class MessageConfigException extends Exception {

    private static final long serialVersionUID = 1L;

    public MessageConfigException(Throwable t) {
	super(t);
    }

    public MessageConfigException(String msg) {
	super(msg);
    }

    public MessageConfigException(String msg, Throwable t) {
	super(msg, t);
    }

}

File: main_project/message_pipe/src/cpp/message_config_exception.cpp
#include "message_config_exception.h"

namespace com {
namespace renren {
namespace messageconfig {

Exception::Exception() : message_("") {

}
Exception::Exception(const std::string& msg) : message_(msg) {

}
const char* Exception::what() const throw() {
  if (message_.empty()) {
    return "empty exception message";
  }
  return message_.c_str();
}

}}} //end com::renren::messageconfig

